src/components/tests/media-briefing/BriefingB.tsx

```diff
@@ -0,0 +1,111 @@
+import React from "react";
+import { Collection as ICollection } from "../../../api";
+import { palette } from "@guardian/src-foundations";
+import { OverlayCard } from "../../cards/OverlayCard";
+import { Multiline } from "../../Multiline";
+import { Heading } from "../../Heading";
+import { HeadlineCard } from "../../cards/HeadlineCard";
+import { Padding } from "../../../layout/Padding";
+import { TableRowCell } from "../../../layout/Table";
+
+export const BriefingB: React.FC<{
+    frontId: string;
+    collections: ICollection[];
+    salt: string;
+}> = ({ collections, salt }) => {
+    const leadStory = collections[0].backfill[0];
+    const topStories = collections[0].backfill.slice(1);
+    const opinionStories = collections[1].backfill;
+    const tvRadioStories = collections[2].backfill;
+    const jobsStories = collections[3].curated;
+
+    const lightGrey = palette.neutral[97];
+    const white = palette.neutral[100];
+    const jobsBackground = "#00A194";
+
+    return (
+        <>
+            <TableRowCell
+                tdStyle={{ backgroundColor: lightGrey, padding: "0" }}
+            >
+                <Multiline topPadding />
+                <Heading heading={collections[0].displayName} />
+
+                <OverlayCard
+                    content={leadStory}
+                    salt={salt}
+                    backgroundColor={white}
+                    layout="compact"
+                />
+
+                <Padding px={12} backgroundColor={palette.neutral[97]} />
+
+                {topStories.map((story, index) => (
+                    <>
+                        {index > 0 && <Padding px={4} />}
+                        <HeadlineCard
+                            content={story}
+                            backgroundColor={white}
+                            showPillarColours
+                            borderWidth="thin"
+                        />
+                    </>
+                ))}
+
+                <Multiline topPadding />
+                <Heading heading={collections[1].displayName} />
+                {opinionStories.map((story, index) => (
+                    <>
+                        {index > 0 && <Padding px={4} />}
+                        <HeadlineCard
+                            content={story}
+                            backgroundColor={white}
+                            showPillarColours
+                            borderWidth="thin"
+                            layout="expanded"
+                        />
+                    </>
+                ))}
+
+                <Multiline topPadding />
+                <Heading heading={collections[2].displayName} />
+                {tvRadioStories.map((story, index) => (
```

Comment:
For this kind of thing, where the is logic in the template, it might be worth pulling it out into a variable - that way the JSX main part keeps a bit smaller and easier to scan/get a sense of. Not always worth it, but can be for cases like this where there are a several mapping/transformation portions in the JSX.

Reply:
Agree. To follow the example from Opinion, I've moved each of these blocks into a separate Collection class, so instead of all of the mappings we'll just have a 'collection' class for each design type.